src/FSharpPlus/Extensions.fs

```diff
@@ -71,6 +71,16 @@ module Result =
             Ok (f x)
         with e -> Error e
 
+    let getOk (source: Result<'T,'Error>) = match source with Ok x -> x | _ -> invalidArg "source" "Result value was Error"
+    let defaultValue value   source = match source with Ok v -> v | _ -> value
+    let defaultWith defThunk source = match source with Ok v -> v | _ -> defThunk ()
```

Comment:
I'd have thought we should use an error mapping function rather than a thunk?
WDYT?
```suggestion
    let defaultWith f source = match source with Ok v -> v | e -> f e
```